File: adventjs/2025/25/solution.ts
function execute(code: string): number {
  const jumps: { [key: number]: number } = {}
  const stack: number[] = []
  const add: { [key: string]: number } = { "+": 1, "-": -1 }
  const isOpen = new Set<string>(["[", "{"])
  const isClose = new Set<string>(["]", "}"])

  for (let i = 0; i < code.length; i++) {
    const c = code[i]
    if (isOpen.has(c)) stack.push(i)
    if (isClose.has(c)) {
      const s = stack.pop()!
      jumps[s] = i + 1
      jumps[i] = s + 1
    }
  }

  let val = 0,
    pos = 0
  while (pos < code.length) {
    const c = code[pos]
    val += add[c] || 0
    const doJump = (isOpen.has(c) && val === 0) || (c === "]" && val !== 0)
    pos = doJump ? jumps[pos] : pos + 1
  }
  return val
}

File: adventjs/2025/25/solution.py
def execute(code: str) -> int:
    jumps = {}
    stack = []
    add = {'+': 1, '-': -1}
    is_open = {'[', '{'}
    is_close = {']', '}'}

    for i, c in enumerate(code):
        if c in is_open:
            stack.append(i)
        if c in is_close:
            s = stack.pop()
            jumps[s] = i + 1
            jumps[i] = s + 1

    val, pos = 0, 0
    while pos < len(code):
        c = code[pos]
        val += add.get(c, 0)
        do_jump = (c in is_open and val == 0) or (c == ']' and val != 0)
        pos = jumps[pos] if do_jump else pos + 1
    return val
